engine/dlib/src/dlib/log.cpp

```diff
@@ -72,10 +72,38 @@ static FILE* g_LogFile = 0;
 static dmCustomLogCallback g_CustomLogCallback = 0;
 static void* g_CustomLogCallbackUserData = 0;
 
+static bool LogAvailable()
+{
+    bool allowed = dLib::IsDebugMode();
+#if  MIN_LEVEL_SEVERITY_DEBUG || MIN_LEVEL_SEVERITY_USER_DEBUG || MIN_LEVEL_SEVERITY_INFO || MIN_LEVEL_SEVERITY_WARNING || MIN_LEVEL_SEVERITY_ERROR || MIN_LEVEL_SEVERITY_FATAL
```

Comment:
Defold defines should be prefixed with "DM_"

Also, I think "MIN_LEVEL" is very non intuitive and not explaining which system they belong to.
Perhaps "DM_LOG_" is a good prefix.

Also, "MIN" implies that there is a "MAX", but there isn't one, so I think we should skip the "MIN" part.